geode-apis-compatible-with-redis/src/test/java/org/apache/geode/redis/internal/data/RedisHashTest.java

```diff
@@ -280,58 +278,39 @@ public void constantBaseRedisHashOverhead_shouldEqualCalculatedOverhead() {
 
   @Test
   public void constantValuePairOverhead_shouldEqualCalculatedOverhead() {
-    int sizeOfDataForOneFieldValuePair = 16; // initial byte[]s are 8 bytes each
-
-    HashMap<ByteArrayWrapper, ByteArrayWrapper> tempHashmap = new HashMap<>();
-
-    ByteArrayWrapper field1 = new ByteArrayWrapper("a".getBytes());
-    ByteArrayWrapper value1 = new ByteArrayWrapper("b".getBytes());
-    ByteArrayWrapper field2 = new ByteArrayWrapper("c".getBytes());
-    ByteArrayWrapper value2 = new ByteArrayWrapper("d".getBytes());
-
-    tempHashmap.put(field1, value1);
-    int oneEntryHashMapSize = reflectionObjectSizer.sizeof(tempHashmap);
-
-    tempHashmap.put(field2, value2);
-    int twoEntriesHashMapSize = reflectionObjectSizer.sizeof(tempHashmap);
-
-    int expectedValuePairOverhead = twoEntriesHashMapSize - oneEntryHashMapSize
-        - sizeOfDataForOneFieldValuePair;
-
-    assertThat(RedisHash.HASH_MAP_VALUE_PAIR_OVERHEAD).isEqualTo(expectedValuePairOverhead);
-  }
-
-  @Test
-  public void constantFirstPairOverhead_shouldEqual_calculatedOverhead() {
-    HashMap<ByteArrayWrapper, ByteArrayWrapper> tempHashmap = new HashMap<>();
-    int emptyHashMapSize = reflectionObjectSizer.sizeof(tempHashmap);
-
-    ByteArrayWrapper field = new ByteArrayWrapper("a".getBytes());
-    ByteArrayWrapper value = new ByteArrayWrapper("b".getBytes());
-
-    tempHashmap.put(field, value);
-    int oneEntryHashMapSize = reflectionObjectSizer.sizeof(tempHashmap);
+    RedisHash hash = new RedisHash(Collections.emptyList());
+    long totalOverhead = 0;
+    final int totalFields = 1000;
+    Random random = new Random(0);
+    for (int fieldCount = 1; fieldCount < totalFields; fieldCount++) {
+      byte[] data = new byte[random.nextInt(30)];
+      random.nextBytes(data);
+      hash.hashPut(data, data);
+      int size = reflectionObjectSizer.sizeof(hash);
+      final int dataSize = 2 * data.length;
+      int overHeadPerField = (size - BASE_REDIS_HASH_OVERHEAD - dataSize) / fieldCount;
+      totalOverhead += overHeadPerField;
+    }
 
-    int expectedFirstPairOverhead = oneEntryHashMapSize - emptyHashMapSize
-        - RedisHash.HASH_MAP_VALUE_PAIR_OVERHEAD;
+    long averageOverhead = totalOverhead / totalFields;
 
-    assertThat(RedisHash.SIZE_OF_OVERHEAD_OF_FIRST_PAIR).isEqualTo(expectedFirstPairOverhead);
+    assertThat(RedisHash.HASH_MAP_VALUE_PAIR_OVERHEAD).isEqualTo(averageOverhead);
```

Comment:
This test fails if you use a value of 2000 for `totalFields`, because the actual value of `averageOverhead` is very close to 48 but can vary slightly to be either greater then or less than 48. With `totalFields` = 1000, the value is 47.931, which is rounded down to 47, but with `totalFields` = 2000, the value is 48.0895, which rounds down to 48. This can be remedied by making `totalOverhead` a float and calling `Math.round(totalOverhead / totalFields)` to determine the value of `averageOverhead`, and changing the value of HASH_MAP_VALUE_PAIR_OVERHEAD to 48.

Reply:
Good point! I've updated the PR to use that approach.